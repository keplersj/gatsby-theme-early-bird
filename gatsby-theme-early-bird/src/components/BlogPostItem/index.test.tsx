import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import { BlogPostItem } from ".";

describe("Blog Post List Item", (): void => {
  beforeAll(() => {
    (useStaticQuery as jest.Mock).mockImplementationOnce((): object => ({
      site: {
        siteMetadata: {
          siteUrl: "http://example.dev"
        }
      }
    }));
  });

  it("renders correctly", (): void => {
    const tree = renderer
      .create(
        <BlogPostItem
          location="/test"
          title="Test Post"
          description="This is a great test post."
          publishDate="December 18, 2018"
          isoDate="2018-12-18"
          wordCount="42"
          minutesNeededToRead="42"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
