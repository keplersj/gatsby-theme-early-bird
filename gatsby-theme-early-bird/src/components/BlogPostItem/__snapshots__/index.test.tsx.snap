// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Post List Item renders correctly 1`] = `
.emotion-1 {
  --starstuff-card-default-background-color: 245,245,245;
  --starstuff-card-default-shadow-color: 10,10,10;
  border-radius: 10px;
  margin: 1em;
  background-color: rgba(var(--starstuff-card-default-background-color),0.5);
  box-shadow: 0.5em 0.5em 0.75em rgba(var(--starstuff-card-default-shadow-color),0.3);
  -webkit-transition: box-shadow 1s;
  transition: box-shadow 1s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (prefers-color-scheme:dark) {
  .emotion-1 {
    --starstuff-card-default-background-color: 55,55,55;
  }
}

.emotion-1:active,
.emotion-1:focus,
.emotion-1:focus-within,
.emotion-1:hover {
  box-shadow: 0.8em 0.8em 1em rgba(var(--starstuff-card-default-shadow-color),0.4);
}

@media screen and (min-width:512px) {
  .emotion-1 {
    min-height: 192px;
  }
}

@media screen and (max-width:512px) {
  .emotion-1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

.emotion-0 {
  margin: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="emotion-1"
>
  <script
    type="application/ld+json"
  >
    Object {
    "@context": "https://schema.org",
    "@id": "http://example.dev/test",
    "@type": "BlogPosting",
    "datePublished": "2018-12-18",
    "headline": "Test Post",
    "mainEntityOfPage": "http://example.dev/test",
    "name": "Test Post",
  }
  </script>
  <div
    className="emotion-0"
  >
    <a
      href="/test"
    >
      <h2>
        Test Post
      </h2>
    </a>
    <span>
      Published 
      December 18, 2018
    </span>
    <p>
      This is a great test post.
    </p>
  </div>
</div>
`;
